Signal/test/phone/PhoneNumberTest.m

```diff
@@ -75,5 +81,78 @@ -(void) testTranslateCursorPosition {
     test([PhoneNumberUtil translateCursorPosition:4 from:@"(5551) 234-567" to:@"(555) 123-4567" stickingRightward:false] == 4);
     test([PhoneNumberUtil translateCursorPosition:5 from:@"(5551) 234-567" to:@"(555) 123-4567" stickingRightward:false] == 7);
 }
+//+ (NSString *)callingCodeFromCountryCode:(NSString *)code;
+//+ (NSString *)countryNameFromCountryCode:(NSString *)code;
+//+ (NSArray *)countryCodesForSearchTerm:(NSString *)searchTerm;
+
+- (void)testCallingCodeFromCountryCode
+{
+    XCTAssertEqualObjects([PhoneNumberUtil callingCodeFromCountryCode:@"US"], @"+1");
+    XCTAssertEqualObjects([PhoneNumberUtil callingCodeFromCountryCode:@"GB"], @"+44");
+    // Invalid country code.
+    XCTAssertEqualObjects([PhoneNumberUtil callingCodeFromCountryCode:@"EK"], @"+0");
+    XCTAssertEqualObjects([PhoneNumberUtil callingCodeFromCountryCode:@"ZZZ"], @"+0");
+    XCTAssertEqualObjects([PhoneNumberUtil callingCodeFromCountryCode:nil], @"+0");
+}
+
+- (void)testCountryNameFromCountryCode
+{
+    XCTAssertEqualObjects([PhoneNumberUtil countryNameFromCountryCode:@"US"], @"United States");
+    XCTAssertEqualObjects([PhoneNumberUtil countryNameFromCountryCode:@"GB"], @"United Kingdom");
+    // Invalid country code.
+    XCTAssertEqualObjects([PhoneNumberUtil countryNameFromCountryCode:@"EK"], @"EK");
+    XCTAssertEqualObjects([PhoneNumberUtil countryNameFromCountryCode:@"ZZZ"], @"ZZZ");
+    testThrows([PhoneNumberUtil countryNameFromCountryCode:nil]);
+}
+
+- (void)testCountryCodesForSearchTerm
+{
+// PhoneNumberUtil needs a valid OWSContactsManager in the TextSecureKitEnv,
+// but we want to avoid constructing the entire app apparatus so we pass nil
+// for the other parameters of the environemtn and disable nonnull warnings.
+#pragma clang diagnostic push
```

Comment:
Take it or leave it:

This is a primary driver for my preference for dependency injection vs. accessing global singletons all over the place.

If you move this test to SSK (where it probably belongs) you can take advantage of a pre-existing menagerie of fakes - e.g. https://github.com/WhisperSystems/SignalServiceKit/blob/master/tests/TestSupport/Fakes/OWSUnitTestEnvironment.m